collectfast/tests/strategies/test_caching_hash_strategy.py

```diff
@@ -57,3 +59,48 @@ def test_gets_and_invalidates_hash(case: TestCase) -> None:
     result_hash = strategy.get_cached_remote_file_hash("path", "prefixed_path")
     case.assertEqual(result_hash, expected_hash)
     mocked.assert_called_once_with("prefixed_path")
+
+
+@make_test
+def test_file_copied_hook_primes_cache(case: TestCase) -> None:
+    clean_static_dir()
+    path = create_static_file()
+    expected_hash = 'abc123'
+    strategy = Strategy()
+    local_storage = FileSystemStorage()
+    with mock.patch.object(strategy, 'get_local_file_hash', return_value=expected_hash) as mock_get_local_file_hash:
+        with mock.patch.object(strategy, 'get_remote_file_hash') as mock_get_remote_file_hash:
+            strategy.file_copied_hook(path.name, path.name, local_storage)
+            actual_hash = strategy.get_cached_remote_file_hash(path.name, path.name)
+            mock_get_remote_file_hash.assert_not_called()
+            mock_get_local_file_hash.assert_called_once()
+    case.assertEqual(actual_hash, expected_hash)
+
+
+@make_test
+@mock.patch("collectfast.strategies.base.HashStrategy.get_local_file_hash")
+def test_get_local_file_hash_memoization(case: TestCase, mocked_super_get_local_file_hash: mock.MagicMock) -> None:
+    foo_hash = 'def456'
+    mocked_super_get_local_file_hash.return_value = foo_hash
+    strategy = Strategy()
+    
+    local_storage = FileSystemStorage()
+    actual_hash = strategy.get_local_file_hash('foo', local_storage)
+    case.assertEqual(actual_hash, foo_hash)
+    
+    actual_hash = strategy.get_local_file_hash('foo', local_storage)
+    case.assertEqual(actual_hash, foo_hash)
+    
+    mocked_super_get_local_file_hash.assert_called_once_with('foo', local_storage)
+    
+    bar_hash = 'ghj789'
+    mocked_super_get_local_file_hash.return_value = bar_hash
+    
+    actual_hash = strategy.get_local_file_hash('bar', local_storage)
+    case.assertEqual(actual_hash, bar_hash)
+    
+    actual_hash = strategy.get_local_file_hash('bar', local_storage)
+    case.assertEqual(actual_hash, bar_hash)
+    
+    mocked_super_get_local_file_hash.assert_called_with('bar', local_storage)
+    case.assertEqual(mocked_super_get_local_file_hash.call_count, 2)
```

Comment:
This is just testing `lru_cache()`. I feel like we don't need test stdlib internals here.

Reply:
It tests that the CachingHashStrategy.get_local_file_hash() method is cached.  The superclass is not.  We are using the `lru_cache()` but it doesn't have to be that way.  This would catch if it was modified in a way that doesn't cache.

Would you like me to remove it?

Reply:
Yes.

Reply:
I've removed it.  Do you want to test that it is decorated with `lru_cache()` or leave it alone?

Reply:
I guess I am quibbling over nothing.  I just checked `sys.getsizeof` on a list containing 5000 string hashes like `hashlib.md5(str(i)).hexdigest()` which is a reasonable number of static files and it is only 43kb.